src/App.tsx

```diff
@@ -1,24 +1,147 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect, useMemo, useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList/TodoList';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { FilteredBy, filteredTodoList } from './helpers';
+import { Errors } from './components/Errors/Errors';
+import { ErrorType } from './types/ErorTypes';
 
-const USER_ID = 0;
+const USER_ID = 12036;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<FilteredBy>(FilteredBy.DefaultType);
+  const [errorType, setErorType] = useState<ErrorType | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [isLoadingTodo, setIsLoadingTodo] = useState<Todo | null>(null);
+
+  const todosToView = useMemo(
+    () => filteredTodoList(todos, filterBy),
+    [todos, filterBy],
+  );
+
+  const todoToRemove = async (todoId: number) => {
+    try {
+      setIsLoadingTodo(todos.find(todo => todo.id === todoId) || null);
```

Comment:
I didn't get it, why do you need this state?
For the future as you need to handle loaders for many todos in the same time you can store array of todos ids that currently have the loading state and show loaders for all needed todos